### PR TITLE
[ios][sharing] Fix crash when share menu is presented on iPad

### LINKED ISSUE #22192: [iPad][expo-sharing] UIPopoverPresentationController should have a non-nil sourceView or barButtonItem set before the presentation occurs
### Minimal reproducible example

https://snack.expo.dev/@honeycomb-wallwisher/26e6b3

### Summary

Here is this error:

```
*** Terminating app due to uncaught exception 'NSGenericException', reason: 'UIPopoverPresentationController (<UIPopoverPresentationController: 0x7f9dfa32d200>) should have a non-nil sourceView or barButtonItem set before the presentation occurs.'
```

I have a patch for this, but the positioning of the popover is not ideal, and will go out of the viewport if the `sourceView` is quite large. Will need the expo team to take it the rest of the way.

```patch
diff --git a/node_modules/expo-sharing/ios/SharingModule.swift b/node_modules/expo-sharing/ios/SharingModule.swift
index 03e85c9..b9403c2 100644
--- a/node_modules/expo-sharing/ios/SharingModule.swift
+++ b/node_modules/expo-sharing/ios/SharingModule.swift
@@ -36,6 +36,12 @@ public final class SharingModule: Module {
         throw MissingCurrentViewControllerException()
       }
         
+      if let popover = activityController.popoverPresentationController {
+        let rootView = currentViewcontroller.view!
+        popover.sourceView = rootView
+        popover.sourceRect = rootView.frame
+      }
+        
       currentViewcontroller.present(activityController, animated: true)
     }
     .runOnQueue(.main)
```

### Environment

```
expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.3.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 16.15.0 - ~/.asdf/installs/nodejs/16.15.0/bin/node
      Yarn: 1.22.19 - ~/.nvm/versions/node/v16.17.1/bin/yarn
      npm: 8.5.5 - ~/.asdf/plugins/nodejs/shims/npm
      Watchman: 2023.01.30.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.12.0 - /Users/thespacemanatee/.asdf/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    IDEs:
      Android Studio: Giraffe 2022.3.1 Canary 11 Giraffe 2022.3.1 Canary 11
      Xcode: 14.3/14E222b - /usr/bin/xcodebuild
    Expo Workflow: bare
```

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why 
`popoverPresentationController` should be configured when on an iPad
Closes #22192

# How
Check if we are on an iPad and configure the `popoverPresentationController`

# Test Plan
Tested on a physical iPad and simulator with bare expo

### CODE REVIEW

File: packages/expo-sharing/ios/SharingModule.swift [hunk 1]
```diff
@@ -36,6 +36,18 @@ public final class SharingModule: Module {
         throw MissingCurrentViewControllerException()
       }
 
+      if UIDevice.current.userInterfaceIdiom == .pad {
+        let viewFrame = currentViewcontroller.view.frame
+        activityController.popoverPresentationController?.sourceRect = CGRect(
```
Comment (tsapeta) [inline packages/expo-sharing/ios/SharingModule.swift:41]: Is setting the `sourceRect` necessary? [Apple docs](https://developer.apple.com/documentation/uikit/uipopoverpresentationcontroller/1622324-sourcerect) states:
> In iOS 13.2 and later, the default value is [CGRectNull](https://developer.apple.com/documentation/coregraphics/cgrectnull), which instructs the system to use the current frame of sourceView. The controller observes changes to this frame and updates the popover accordingly

So I'm guessing it should work fine enough 🤔

Comment (alanjhughes) [reply packages/expo-sharing/ios/SharingModule.swift:41]: This is the error without this
`*** Terminating app due to uncaught exception ‘NSGenericException’, reason: ‘Your application has presented a UIActivityViewController (<UIActivityViewController: 0x7f829b854600>). In its current trait environment, the modalPresentationStyle of a UIActivityViewController with this style is UIModalPresentationPopover. You must provide location information for this popover through the view controller’s popoverPresentationController. You must provide either a sourceView and sourceRect or a barButtonItem. If this information is not known when you present the view controller, you may provide it in the UIPopoverPresentationControllerDelegate method -prepareForPopoverPresentation.’`

sourceRect or sourceView needs to be specified for the popover to be visible. sourceView isn't enough in our case. The view still does not appear so I used the same approach as [here](https://github.com/expo/expo/blob/9ddb76441ee079486205de8a311800ff6618773e/packages/expo-document-picker/ios/DocumentPickerModule.swift#L35)

File: packages/expo-sharing/ios/SharingModule.swift [hunk 2]
```diff
@@ -36,6 +36,18 @@ public final class SharingModule: Module {
         throw MissingCurrentViewControllerException()
       }
 
+      if UIDevice.current.userInterfaceIdiom == .pad {
```
[the remark this answers is not in the document]
Comment (alanjhughes) [reply packages/expo-sharing/ios/SharingModule.swift:39]: Will do. It's mentioned in the [overview](https://developer.apple.com/documentation/uikit/uiactivityviewcontroller#overview) in the docs so I'll add that

**Review by tsapeta — APPROVED** [review]

**Comment by thespacemanatee** [comment]
Is it possible to release this since it's a launch-blocking bug? Thanks team!

**Comment by tsapeta** [comment]
It's out in `11.3.2` 😉

**Pull request merged**